### ISSUE TITLE
Leading space is incorrectly replaced in dom elements

### ISSUE THREAD

**Issue by otonashixav** [solidjs/solid#1367]
### Describe the bug

When a dom element's children consists of a leading space then some dynamic content, the space is removed.

### Your Example Website or App

https://playground.solidjs.com/anonymous/fc437f14-a6f6-4613-b37c-1e48f6b9be6d

### Steps to Reproduce the Bug or Issue

1. Open the playground link

### Expected behavior

The space should stay. 

### Screenshots or Videos
```css
button {
    white-space: pre;
} 
```
```jsx
<div><button> {(() => 1)()}</button> - Problem</div>
<div><button> {1}</button> - Expected</div>
```

![image](https://user-images.githubusercontent.com/42372774/202267133-d4d96e03-26ea-4eb0-bd15-ddea07e83f55.png)

### Platform

I've only tested it in the playground. Sorry if this was already fixed.

### Additional context

This seems to be a compilation oversight? If the leading space is replaced with a different character the call to `insert` in the compiled output has an extra third parameter `null`. Also this is fine (space doesn't get removed).
```jsx
<button>
  {" "}
  {(() => 1)()}
</button>
```

**Comment by fromaline** [comment]
I investigated this issue, as far as I see, it's an error inside [babel-plugin-jsx-dom-expressions](https://github.com/ryansolid/dom-expressions/tree/main/packages/babel-plugin-jsx-dom-expressions).
It indeed handles the leading space incorrectly and doesn't replace dynamic content with the `<!>` in the template (Interestingly if you add extra space at the end, it works as expected). Then [`insertExpression` just overwrites the leading space with `parent.textContent = value`](https://github.com/ryansolid/dom-expressions/blob/main/packages/dom-expressions/src/client.js#L394)

I'll try to work on this and maybe fix it.

**Comment by ryansolid** [comment]
Yeah, this is a bit tricky. There are a couple of ways to solve this. Technically speaking it should be treated as a multi insert but white space removal is messing with it. However messing with whatever logic caused white space removal to work like that is probably going to touch other places in the code. I'm going to review and see why I chose not to count white spaces towards multi as that is an interesting choice that impacts multiple things.

**Comment by fromaline** [comment]
@ryansolid, multi insert is something like this?

```js
_$insert(_el$, () => 1, _el$2);
_$insert(_el$, () => " ", null);
```

If yes, why it should be treated as a multi insert? With extra space after the expression, it works perfectly fine:

```js
const _tmpl$ = /*#__PURE__*/_$template(`<button> <!> </button>`, 3);
```

**Comment by ryansolid** [comment]
Ok This has been fixed in babel-plugin-jsx-dom-expressions 0.35.6. I will bump Solid preset in the next release. But clearing your package lock and a fresh install should have yo fixed.
